prov/efa/src/rxr/rxr_msg.c

```diff
@@ -786,6 +787,55 @@ struct rxr_rx_entry *rxr_msg_split_rx_entry(struct rxr_ep *ep,
 	return rx_entry;
 }
 
+/**
+ * @brief find an unexpected rx entry for a receive operation.
+ *
+ * @param[in]	ep	endpoint
+ * @param[in]	addr	fi_addr of the peer want to receive from, can be FI_ADDR_UNSPEC
+ * @param[in]	tag	tag of the unexpected message, used only if op is ofi_op_tagged.
+ * @param[in]	ignore	mask of the tag, used only if op is ofi_op_tagged.
+ * @param[in]	op	either ofi_op_tagged or ofi_op_msg.
+ * @return	If an unexpected rx_entry was found, return the pointer.
+ * 		Otherwise, return NULL.
+ */
+static inline
+struct rxr_rx_entry *rxr_msg_find_unexp_rx_entry(struct rxr_ep *ep, fi_addr_t addr,
+						 int64_t tag, uint64_t ignore, uint32_t op)
+{
+	struct rxr_match_info match_info;
+	struct dlist_entry *match;
+	struct rdm_peer *peer;
+
+	peer = (ep->util_ep.caps & FI_DIRECTED_RECV) ? rxr_ep_get_peer(ep, addr) : NULL;
+
+	if (op == ofi_op_msg) {
+		if (peer) {
+			match = dlist_empty(&peer->rx_unexp_list) ? NULL : peer->rx_unexp_list.next;
+			return match ? container_of(match, struct rxr_rx_entry, peer_unexp_entry) : NULL;
+		}
+
+		match = dlist_empty(&ep->rx_unexp_list) ? NULL : ep->rx_unexp_list.next;
+		return match ? container_of(match, struct rxr_rx_entry, entry) : NULL;
+	}
+
+	assert(op == ofi_op_tagged);
+	match_info.tag = tag;
+	match_info.ignore = ignore;
+
+	if (peer) {
+		match = dlist_find_first_match(&peer->rx_unexp_tagged_list,
+		                               rxr_msg_match_peer_unexp_by_tag,
+					       (void *)&match_info);
+		return match ? container_of(match, struct rxr_rx_entry, peer_unexp_entry) : NULL;
+	}
+
+	match = dlist_find_first_match(&peer->rx_unexp_tagged_list,
+				       rxr_msg_match_ep_unexp_by_tag,
+				       (void *)&match_info);
+
```

Comment:
you are inconsistent with your whitespace.  In the peer case, you do not have a blank line between the search and the return and here you do.